index.d.cts

```diff
@@ -0,0 +1,54 @@
+declare const _exports: PrettierSync;
+export = _exports;
+export type Prettier = typeof import("prettier");
```

Comment:
Is it possible to do something like

```js
import type Prettier from "prettier";
```

?

Reply:
Let me look into it

It's the `tsc` compiler automatically exporting `@typedef` by default

Reply:
How about hand-writing code like this

```ts
import type {
  version,
  util,
  doc,
  formatWithCursor,
  format,
  check,
  resolveConfig,
  resolveConfigFile,
  getFileInfo,
  getSupportInfo,
} from "prettier";

type SynchronizedPrettier = {
  // Prettier static properties
  version: typeof version;
  util: typeof util;
  doc: typeof doc;

  // Prettier functions
  formatWithCursor: Awaited<ReturnType<typeof formatWithCursor>>;
  format: Awaited<ReturnType<typeof format>>;
  check: Awaited<ReturnType<typeof check>>;
  resolveConfig: Awaited<ReturnType<typeof resolveConfig>>;
  resolveConfigFile: Awaited<ReturnType<typeof resolveConfigFile>>;
  getFileInfo: Awaited<ReturnType<typeof getFileInfo>>;
  getSupportInfo: Awaited<ReturnType<typeof getSupportInfo>>;
};
```

Reply:
Entirely up to you 😊

With JSDoc and `checkJs` enabled (without switching to TypeScript) you'll be able to use code completion, see documentation in tooltips, and spot and errors and warnings as-you-type

It lets the compiler validate that type declarations are correct when [`prettier` `"./src/index.d.ts"` types](https://github.com/prettier/prettier/blob/main/package.json#L17C17-L17C33) change without having to keep types consistent by hand

### tsconfig.json
```json
{
  "compilerOptions": {
    "allowJs": true,
    "checkJs": true
  }
}
```

I'd be more than happy if you used this PR for guidance to add them manually 👍

Let me know if you'd like anything splitting out